test-automation/staking-standalone-tests/staking-test.py

```diff
@@ -0,0 +1,238 @@
+    CLI.remove_account(del_acc_name)
+    CLI.single_call(f"hmy keys add {del_acc_name}")
+    init_del_funds = DELEGATE + random.randint(0, int(MAX_TOTAL_DELEGATION - MIN_SELF_DELEGATION - DELEGATE - AMOUNT))
+    init_val_funds = random.randint(max(AMOUNT, MIN_SELF_DELEGATION), int(MAX_TOTAL_DELEGATION - init_del_funds))
+    init_del_funds += GAS_OVERHEAD
+    init_val_funds += GAS_OVERHEAD
+    assert init_del_funds + init_val_funds < MAX_TOTAL_DELEGATION + 2 * GAS_OVERHEAD
+    fund_account(funding_account, val_acc_name, init_val_funds)
+    fund_account(funding_account, del_acc_name, init_del_funds)
+    return val_acc_name, del_acc_name
+
+
+@test
+def staking_delegation(val_name, del_name):
+    assert check_balance(val_name)[0]["amount"] >= max(AMOUNT, MIN_SELF_DELEGATION)
+    assert check_balance(del_name)[0]["amount"] >= DELEGATE
+
+    val_address = CLI.get_address(val_name)
+    del_address = CLI.get_address(del_name)
+    SAVED_ITEM_PATHS.append(f"{CLI.keystore_path}/{val_name}")
+    SAVED_ITEM_PATHS.append(f"{CLI.keystore_path}/{del_name}")
+    print(f"{COLOR.HEADER}Validator address: {val_address}")
+    print(f"Delegator address: {del_address}{COLOR.ENDC}\n")
+
+    # Create a new validator with a fresh BLS key.
+    shard_count = len(get_sharding_structure(args.endpoint)['result'])
+    bls_key = list(bls_generator(1, os.getcwd(), filter_fn=lambda k: int(k["public-key"], 16) % shard_count == 0))[0]
+    bls_key_path = bls_key["encrypted-private-key-path"]
+    SAVED_ITEM_PATHS.append(bls_key_path)
+    print(f"{COLOR.OKGREEN}New BLS key:{COLOR.ENDC}\n{json.dumps(bls_key, indent=4)}\n")
+    proc = CLI.expect_call(f"hmy --node={args.endpoint} staking create-validator --validator-addr {val_address} "
+                           f"--name {val_name} --identity test_account --website harmony.one "
+                           f"--security-contact Daniel-VDM --details none --rate 0.1 --max-rate 0.9 "
+                           f"--max-change-rate 0.05 --min-self-delegation {MIN_SELF_DELEGATION} "
+                           f"--max-total-delegation {MAX_TOTAL_DELEGATION} "
+                           f"--amount {AMOUNT} --bls-pubkeys {bls_key['public-key']} "
+                           f"--chain-id {args.chain_id}")
+    proc.expect("Enter the absolute path to the encrypted bls private key file:\r\n")
+    proc.sendline(bls_key_path)
+    proc.expect("Enter the bls passphrase:\r\n")
+    proc.sendline("")
+    proc.expect("Repeat the bls passphrase:\r\n")
+    proc.sendline("")
+    txn = json.loads(proc.read().decode().replace('\r', '').replace('\n', ''))
+    print(f"{COLOR.OKGREEN}Created validator {val_address}:{COLOR.ENDC}\n{json.dumps(txn, indent=4)}\n")
+    print(f"{COLOR.OKBLUE}Sleeping 25 seconds for finality...{COLOR.ENDC}\n")
+    time.sleep(25)
+
+    # Check the create-validator staking transaction.
+    transaction = json.loads(CLI.single_call(f"hmy --node={args.endpoint} blockchain "
```

Comment:
staking transactions now support wait-for-confirm. should you use that instead of this?